### PR TITLE
Plus2 for non mods

### PR DESCRIPTION
**Pull request by Des-**
Solves #10 

+ Please check sql query in zone/matches ~line 906
+ Please check if it works in your sandbox. I am still getting 100.000 Warnings in the frontend, but I think it is related to my corruped installation and not to the code. \eruschwein.
+ damn....I hope you have reverted some of my commits. I have no idea why our master is still behind my hopefullyEnoughCiv commits. The first 5 commits should already be in our master. :( Dunno if it destroys anything.^^
+ Be careful when putting this live. There is a lot of danger potential, because i touched every method in the api.

### CODE REVIEW

File: zone/matches.php [hunk 1]
```diff
@@ -899,4 +899,14 @@ public function draw(int $user_id): array
             return $match_ids;
         });
     }
+
+    //TODO: eru....can we do it like that? This looks way easier than the complicated sql stuff you are doing....
+    public function get_drawer_id(int $match_id): int
```
Comment (mcepok) [inline zone/matches.php]: Please rename this to `get_draw_user_id`

File: zone/matches.php [hunk 2]
```diff
@@ -899,4 +899,14 @@ public function draw(int $user_id): array
             return $match_ids;
         });
     }
+
+    //TODO: eru....can we do it like that? This looks way easier than the complicated sql stuff you are doing....
```
Comment (mcepok) [inline zone/matches.php]: Looks fine to me, but I don't know what complicated sql stuff you're referring to.

File: controller/api.php [hunk 3]
```diff
@@ -446,19 +463,16 @@ public function match_post_result(int $match_id): JsonResponse
...
             'match_id' => $match_id,
         ]);
     }
 
     public function information(): JsonResponse
     {
         return $this->respond(function () {
-            if (!$this->auth->acl_get(acl::u_zone_view_info)) { // todo: this does not work
-                return [];
+            if (!acl::has_permission($this->auth, self::is_activated($this->get_user_id()), acl::u_zone_view_info)) {
```
[the remark this answers is not in the document]
Comment (Des-) [reply controller/api.php]: I am not sure what it does. I don't have the max important information feature in my sandbox. Maybe it fixes it, maybe not.^^

File: controller/api.php [hunk 4]
```diff
@@ -275,15 +279,19 @@ public function replace_confirm(int $replace_user_id): JsonResponse
...
-            acl::m_zone_change_match => 'NCZONE_REASON_NOT_ALLOWED_TO_DRAW',
-        ], [
             'replace_user_id' => $replace_user_id,
         ]);
     }
 
     public function add_pair_preview(int $match_id): JsonResponse
     {
         return $this->respond(function ($args) {
+            $allowed_to_do_this = false;
```
[the remark this answers is not in the document]
Comment (Des-) [reply controller/api.php]: I think it is way easier to read with the extra variable that I introduced here. But yes, we can join both in a very long and complicated 'if' if you like it more.

File: controller/api.php [hunk 5]
```diff
@@ -56,11 +55,6 @@ private function respond(
         }
 
         try {
-            foreach ($acl as $perm => $err) {
```
[the remark this answers is not in the document]
Comment (Des-) [reply controller/api.php]: done. (not in replace functions because we want to modify them anyways)

**Pull request merged**